### PR TITLE
Update ESLint to 1.0.0

### LINKED ISSUE #264: Update ESLint to 1.0.0rc1
http://eslint.org/blog/2015/07/eslint-1.0.0-rc-1-released/

### PR DESCRIPTION
**Pull request by saneyuki**
Fix #264

[Review on Reviewable](https://reviewable.io/reviews/karen-irc/karen/265)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -72,18 +77,29 @@
...
         }],
+        "init-declarations": [2, "always"],
+        "no-class-assign": 2,
+        "no-const-assign": 2,
+        "no-invalid-this": 1,
         "no-this-before-super": 2,
         "no-var": 1,
         "object-shorthand": 0,
         "prefer-const": 1,
+        "prefer-reflect": 1,
```
Comment (saneyuki) [inline .eslintrc]: @Constellation how do you feel [this rule](http://eslint.org/docs/1.0.0/rules/prefer-reflect)? I doubt we really need to enable this.

**Comment by dokidokivisual** [comment]
:umbrella: The latest upstream changes (presumably #281) made this pull request unmergeable. Please resolve the merge conflicts.

**Comment by Constellation** [comment]
@dokidokivisual r+

---

Review status: 0 of 6 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[.eslintrc, line 100 \[r1\]](https://reviewable.io:443/reviews/karen-irc/karen/265#-Jw65IEUR4Y-rdx125D6)** ([raw file](https://github.com/karen-irc/karen/blob/0b0b4cd95568b5c7e8c75c191a56b24520851358/.eslintrc#L100)):</sup>
I'm not confident that using `Reflect` instead of `Object` is so important.
But, `Reflect` namespace explicitly says "exposes Reflection functionality". So, I think it's ok.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/karen-irc/karen/265)

**Comment by dokidokivisual** [comment]
:pushpin: Commit e4ed058 has been approved by `Constellation`

**Comment by dokidokivisual** [comment]
:hourglass: Testing commit e4ed058 with merge f062b30...

**Comment by dokidokivisual** [comment]
:sunny: Test successful - [travis](https://travis-ci.org/karen-irc/karen/builds/74529897)

**Pull request merged**